torch/csrc/jit/passes/onnx/peephole.cpp

```diff
@@ -637,11 +637,22 @@ static void fuseListConstructListUnpack(Block* b) {
     for (auto* child_block : it->blocks()) {
       fuseListConstructListUnpack(child_block);
     }
-    if (it->kind() == prim::ListUnpack &&
-        it->input()->node()->kind() == prim::ListConstruct) {
+    if (it->kind() == prim::ListUnpack) {
       for (size_t i = 0; i < it->outputs().size(); i++) {
         auto output = it->outputs().at(i);
-        output->replaceAllUsesWith(it->input()->node()->inputs().at(i));
+        if (it->input()->node()->kind() == prim::ListConstruct) {
+          output->replaceAllUsesWith(it->input()->node()->inputs().at(i));
+        } else if (it->input()->type()->cast<ListType>()) {
```

Comment:
Should this be separated into a function on it's own? The input is not ListConstruct, so not matching up to the method name actually.

Reply:
yes, I was thinking it's maybe better to change the name of the peephole pass instead